src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -274,6 +274,7 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
 
     if (MPIDI_OFI_ENABLE_DATA) {
         hints->caps |= FI_DIRECTED_RECV;        /* Match source address    */
+        hints->domain_attr->cq_data_size = 4;   /* Minimum size for completion data entry */
```

Comment:
So in your comment above, it looks like you're overwriting the value for `ENABLE_DATA` to say that it's enabled. If the provider only supports a `cq_data_size` of 0, then it can't really be enabled. The `cq_data_size` field is what we use for the immediate data so all of the `fi_tinjectdata`/`fi_tsenddata` calls use it.

The change in this patch is so we don't require 8 bytes for that field since we were accidentally using `uint64_t` for the datatype there instead of `uint32_t`.